### PR TITLE
App Templates

### PR DESCRIPTION
**Pull request by skellock**
### What Is This?

App templates (i'm open to changing this name).

This gives us a way to seed a fresh react native project.

### Why?

We've been using `ignite-basic-structure` up to this point.  Now that React Native core is using templates, we should embrace that too.  The benefit is: people can now customize their new projects. Bring Your Own Opinions.

We're still going to rock the kazbah out of our own Infinite Red based app templates (after all, that's what Ignite is to us), but you still can use `ignite` your way.

### What Changes?

Externally nothing just yet.  There will be a `ignite n MyApp --template BringingMySquad` soon.

Internally, we now have 2 app templates now:

* `ignite-unholy-app-template` - really, it's our Ignite 1 opinions but modular.  we're still in the process of ripping this apart.
* `ignite-minimal-app-template` - "some" of Ignite 1's opinion, but with way way way less.

(I personally would love to see an `none` option too.)

### CODE REVIEW

**Comment by jamonholmgren** [comment]
@skellock This is headed the right direction. :100:

**Comment by skellock** [comment]
Thx dude.  Stage 2 won't be much work either.

**Comment by skellock** [comment]
This PR got whacked by the copy screen merge.  I'll rebase this guy tomorrow morning.  Walking Dead is on in 20 mins!

**Comment by jamonholmgren** [comment]
Looks amazing -- @skellock, you're 7 minutes into zombieland, but when you see this, 👍

**Comment by skellock** [comment]
Ok.  PR is ready to go!

**Comment by GantMan** [comment]
This is coming out awesome!!!!

**Comment by skellock** [comment]
NO YOU'RE COMING OUT AWESOME!

File: packages/ignite/src/extensions/reactNative.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    // jet if the version isn't available
+    const versionCheck = await system.run(`npm info react-native@${reactNativeVersion}`)
+    const versionAvailable = test(new RegExp(reactNativeVersion, ''), versionCheck || '')
+    if (!versionAvailable) {
+      print.error(`💩  react native version ${print.colors.yellow(reactNativeVersion)} not found on NPM -- ${print.colors.yellow(REACT_NATIVE_VERSION)} recommended`)
+      return exitCodes.REACT_NATIVE_VERSION
+    }
+
+    // ok, let's do this
+    const spinner = print.spin(`adding ${print.colors.cyan('React Native ' + reactNativeVersion)} ${print.colors.muted('(~60 seconds)')}`)
```
Comment (GantMan) [inline packages/ignite/src/extensions/reactNative.js]: I miss `ish` at the end of an already ish.   Always made me smile.

**Pull request merged**